scripts/compile.js

```diff
@@ -0,0 +1,331 @@
+/* eslint-disable */
+const Vash = require('vash');
+const fs = require('fs');
+const Path = require('path');
+const Url = require('url');
+const Gettext = require('./gettext');
+const Crypto = require('crypto');
+const common = require('./common');
+const generate_static_data = require('./generate-static-data');
+const program = require('commander');
+
+const CONTENT_PLACEHOLDER = "CONTENT_PLACEHOLDER"; // used in layout.vash
+
+program
+    .version('0.1.0')
+    .description('Build .vash templates into /dist folder')
+    .option('-d, --dev', 'Build for your gh-pages')
+    .option('-b, --branch [branchname]', 'Build your changes to a sub-folder named: br_branchname')
+    .option('-p, --path [save_as]', 'Re-compile only the template with save_as in its URL')
+    .option('-t, --add-translations', 'Update messages.pot with new translations')
+    .parse(process.argv);
+
+/**********************************************************************
+ * Vash helpers
+ **********************************************************************/
+
+global.view = { };
+Vash.helpers.when = (condition, value) => {
+    if(!condition) { return undefined; }
+    return Vash.helpers.raw(value);
+}
+Vash.helpers.class = value =>  Vash.helpers.when(value, `class="${value}"`);
+Vash.helpers.id    = value =>  Vash.helpers.when(value, `id="${value}"`);
+Vash.config.debug  = true;
+Vash.config.debugparser = true;
+Vash.config.debugcompiler = true;
+Vash.config.cache = true;
+Vash.config.settings = {
+    views: Path.join(common.root_path, 'src/templates')
+}
+
+/**********************************************************************
+ * Common functions
+ **********************************************************************/
+function getConfig() {
+    const config = {
+        languages : common.languages,
+        sections  : ['app', 'static'],
+        root_path : common.root_path,
+        add_translations: false,
+    }
+    if (program.branch === 'translations') {
+        config.languages = ['ACH'];
+    }
+    config.dist_path = Path.join(config.root_path, 'dist');
```

Comment:
`branch` might be missing here.

Reply:
When `branch` is missing then `config.languages` will remain as `common.languages` which has the list of all languages.

Reply:
If `branch` is given, does it make the directories as `dist/[branch]/[lang]/...`?